lib/puppet/application/resource.rb

```diff
@@ -136,14 +136,16 @@ def main
     resources = find_or_save_resources(type, name, params)
 
     if options[:to_yaml]
-      text = resources.
-        map { |resource| resource.prune_parameters(:parameters_to_include => @extra_params).to_hierayaml }.
-        join("\n")
+      text = resources.map do |resource|
+        r = resource.prune_parameters(:parameters_to_include => @extra_params).to_hierayaml
+        r.encoding == Encoding.default_external? ? r : r.force_encoding(Encoding.default_external)
```

Comment:
Can this just be simplified to `r.force_encoding(Encoding.default_external)` - i.e. is the check roughly as costly as performing the effectual `no-op`?